src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -481,6 +492,27 @@ internal static bool TryGetAnnotation(this IEdmModel model, IEdmElement annotata
             return val;
         }
 
+        /// <summary>
+        /// Gets full name for the schema element with the provided namespace and name
+        /// </summary>
+        /// <param name="elementNamespace">Namespace of the element</param>
+        /// <param name="elementName">The element name</param>
+        /// <returns>The full name of the element</returns>
+        internal static string GetFullNameForSchemaElement(string elementNamespace, string elementName)
+        {
+            if (elementName == null)
+            {
+                return string.Empty;
+            }
+
+            if (elementNamespace == null)
+            {
+                return elementName;
+            }
+
+            return elementNamespace + "." + elementName;

```

Comment:
At least here we need to do empty-string-test for elementNamespace, otherwise could get sth like ".MyElemName"?

Reply:
Gist of my changes if performance. So I am not comfortable making changes that lead to any functional changes. For the large part I am replicating the exact code that existed before and instead of calculating the full name every time, I am caching it on the object and reusing it over and again.


I do not like to mix and match perf and functional changes. I can open a new issue for you to look at this.

Reply:
Yes; please open an issue.  Agree that this looks suspect, but also agree that this is exactly mimicking current behavior. 

---
In reply to: [209781364](https://github.com/OData/odata.net/pull/1227#discussion_r209781364) [](ancestors = 209781364)

Reply:
Done: https://github.com/OData/odata.net/issues/1248

Reply:
Thanks!

---
In reply to: [216468441](https://github.com/OData/odata.net/pull/1227#discussion_r216468441) [](ancestors = 216468441)